lightningd/gossip_control.c

```diff
@@ -27,69 +27,34 @@ static void gossip_finished(struct subd *gossip, int status)
 	errx(1, "Gossip failed (signal %u), exiting.", WTERMSIG(status));
 }
 
-static void peer_bad_message(struct subd *gossip, const u8 *msg)
-{
-	u64 unique_id;
-	struct peer *peer;
-	u8 *err;
-
-	if (!fromwire_gossipstatus_peer_bad_msg(msg, msg, NULL,
-						&unique_id, &err))
-		fatal("Gossip gave bad PEER_BAD message %s", tal_hex(msg, msg));
-
-	peer = peer_by_unique_id(gossip->ld, unique_id);
-	if (!peer)
-		fatal("Gossip gave bad peerid %"PRIu64, unique_id);
-
-	log_debug(gossip->log, "Peer %s gave bad msg %s",
-		  type_to_string(msg, struct pubkey, &peer->id),
-		  tal_hex(msg, msg));
-	peer_fail_permanent(peer, msg);
-}
-
-static void peer_failed(struct subd *gossip, const u8 *msg)
-{
-	u64 unique_id;
-	struct peer *peer;
-	u8 *err;
-
-	if (!fromwire_gossipstatus_peer_failed(msg, msg, NULL,
-					       &unique_id, &err))
-		fatal("Gossip gave bad PEER_FAILED message %s",
-		      tal_hex(msg, msg));
-
-	peer = peer_by_unique_id(gossip->ld, unique_id);
-	if (!peer)
-		fatal("Gossip gave bad peerid %"PRIu64, unique_id);
-
-	peer_fail_permanent(peer, msg);
-}
-
 static void peer_nongossip(struct subd *gossip, const u8 *msg,
 			   int peer_fd, int gossip_fd)
 {
-	u64 unique_id;
-	struct peer *peer;
-	u8 *inner;
+	struct pubkey id;
 	struct crypto_state cs;
-
-	if (!fromwire_gossipstatus_peer_nongossip(msg, msg, NULL,
-						  &unique_id, &cs, &inner))
-		fatal("Gossip gave bad PEER_NONGOSSIP message %s",
+	u8 *gfeatures, *lfeatures, *in_pkt;
+
+	if (!fromwire_gossip_peer_nongossip(msg, msg, NULL,
+					    &id, &cs,
+					    &gfeatures,
+					    &lfeatures,
+					    &in_pkt))
+		fatal("Gossip gave bad GOSSIP_PEER_NONGOSSIP message %s",
 		      tal_hex(msg, msg));
 
-	peer = peer_by_unique_id(gossip->ld, unique_id);
-	if (!peer)
-		fatal("Gossip gave bad peerid %"PRIu64, unique_id);
-
-	if (peer->owner != gossip)
-		fatal("Gossip gave bad peerid %"PRIu64" (owner %s)",
-		      unique_id, peer->owner ? peer->owner->name : "(none)");
-
-	log_info(peer->log, "Gossip ended up receipt of %s",
-		 wire_type_name(fromwire_peektype(inner)));
+	/* We already checked the features when it first connected. */
```

Comment:
So why are we checking them here again? A short hint in the comment would be nice.